### PR TITLE
Add ChefCorrectness/InvalidDefaultAction cop

### PR DESCRIPTION
**Pull request by tas50**
Detect when people are setting it to something that isn't a variable, symbol, or array.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: config/cookstyle.yml [hunk 1]
```diff
@@ -454,6 +454,15 @@ ChefCorrectness/InvalidPlatformInCase:
     - '**/metadata.rb'
     - '**/Berksfile'
 
+ChefCorrectness/InvalidDefaultAction:
+  Description: Default actions in resources should be symbols or an array of symbols.
```
Comment (phiggins) [inline config/cookstyle.yml:458]: Maybe this is better?

```suggestion
  Description: Default actions in resources should be a symbol or an array of symbols.
```

File: spec/rubocop/cop/chef/correctness/invalid_default_action_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  it 'does not register an offense with a symbol' do
+    expect_no_offenses('default_action :create')
+  end
+
+  it 'does not register an offense with a variable' do
+    expect_no_offenses('default_action foo')
+  end
+
+  it 'does not register an offense with a variable' do
```
Comment (phiggins) [inline spec/rubocop/cop/chef/correctness/invalid_default_action_spec.rb:39]:
```suggestion
  it 'does not register an offense with an array of symbols' do
```

File: spec/rubocop/cop/chef/correctness/invalid_default_action_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    expect_no_offenses('default_action [:create, :enable]')
+  end
+end
```
Comment (phiggins) [inline spec/rubocop/cop/chef/correctness/invalid_default_action_spec.rb:42]: I might add a test for an array of non-symbols.

Comment (tas50) [reply spec/rubocop/cop/chef/correctness/invalid_default_action_spec.rb:42]: Right now I just sorta pass on detecting that. Any array is fine. I saw my first case of someone using that while writing this and I don't think anyone really realizes it's even an option.

**Pull request merged**